static/js/subs.js

```diff
@@ -743,19 +742,27 @@ $(function () {
     $("#stream_creation_form").on("submit", function (e) {
         e.preventDefault();
         var stream = $.trim($("#create_stream_name").val());
-        var principals = _.map(
-            $("#stream_creation_form input:checkbox[name=user]:checked"),
-            function (elem) {
-                return $(elem).val();
-            }
-        );
-        // You are always subscribed to streams you create.
-        principals.push(page_params.email);
-        ajaxSubscribeForCreation(stream,
-            principals,
-            $('#stream_creation_form input[name=privacy]:checked').val() === "invite-only",
-            $('#announce-new-stream input').prop('checked')
+        var stream_status = compose.check_stream_existence(stream);
```

Comment:
We shouldn't be using this **synchronous** check for whether the stream exists; sync queries are problematic and there's a race here.  If we want to do something different in the event that the stream already exists, we should pass an argument to the subscribe call here indicating we want to create a new stream and it should throw an error if the stream already exists (the backend doesn't support this, I think, but it'd be easy to add).